src/calendar/calendar-month.tsx

```diff
@@ -47,7 +47,18 @@ export const CalendarMonth = ({
     };
 
     const handleMonthClick = (value: Dayjs) => {
-        onSelect(value);
+        let result = value;
+
+        if (selectedStartDate) {
+            const [yyyy, , dd] = dayjs(selectedStartDate)
+                .format("YYYY-MM-DD")
+                .split("-");
+            const selectedMonth = value.format("MM");
+
+            result = dayjs(`${yyyy}-${selectedMonth}-${dd}`);
+        }
+
+        onSelect(result);
```

Comment:
is this bug because generateMonths and generateDecadeOfYears use calendarDate instead of selectedStartDate?

Reply:
Yes, kind of.

We have two scenario:
1. User didn't select day in calendar but selected Month or Year first -> we need to selected default date and update accordingly if user update for month.
2. User selected day and update month calendar, so need to know some value been selected.

Reply:
I'll update the year calendar fn next day. I spotted one bug. 

Selected 2024 Feb 29 -> update to -> 2023 Feb -> it is not selected last day of Feb.

Reply:
ok. I think instead of fixing it in handleMonthClick, should fix the date generation instead

Reply:
Thanks giving some clue. let me think. :))

Reply:
`generateMonths` which a function to generate calendar months.

All month value will be same as selected value in the calendar for prev/next year if we update the selected value inside this value.